### PR TITLE
fixed multi lines under list item issue

### PR DESCRIPTION
**Pull request by donganzh**
@azu Previously, the maintainer tried to fix the issue about multi lines in a list item but it still didn't work. 
Therefore, I am proposing a new fix of this issue and I add the test case in the test.js file. Please take a look. Here is my test case which passed. Thanks for the tool!
```
- list item 1

    1. A paragraph describing this list item 1
    
        this is a subparagraph.
        
    2.  paragraph 2
    
        this is a new paragraph.

- list item 2

    A paragraph describing this list item 2.

- list item 3

    Another paragraph describing list item 3.
```

fix #5

### CODE REVIEW

File: test/textlint-rule-period-in-list-item-test.js [hunk 1]
```diff
@@ -63,9 +83,9 @@ tester.run("textlint-rule-period-in-list-item", rule, {
         // allowOrderedList
         {
             text: `
-1. item1.
-2. item2.
-3. item3.
+1. item1
```
Comment (azu) [inline test/textlint-rule-period-in-list-item-test.js]: This test case is for `allowOrderedList` option.
Original test case is valid

File: test/textlint-rule-period-in-list-item-test.js [hunk 2]
```diff
@@ -30,7 +30,8 @@ tester.run("textlint-rule-period-in-list-item", rule, {
         {
             text: `- [text](http://example.com)`,
             options: {
-                periodMark: ""
+                periodMark: "",
+                ignoreOnlyLink: false
```
Comment (azu) [inline test/textlint-rule-period-in-list-item-test.js]: Why is it changed?

We should add new test case for this PR.

File: test/textlint-rule-period-in-list-item-test.js (hunk 1 above)
Comment (donganzh) [reply test/textlint-rule-period-in-list-item-test.js]: ok, let’s keep it

**Comment by donganzh** [comment]
@azu Morning! I tested my code with all your test cases and they all passed. I am applying this tool to multiple documents from many repos so it would be nice if we have this issue fixed. I think the main issue is to get the first paragraph under children under ListItem, since this rule we only need to check if the list sentence has a period or not, the paragraph below it doesn't matter.

File: test/textlint-rule-period-in-list-item-test.js (hunk 2 above)
Comment (donganzh) [reply test/textlint-rule-period-in-list-item-test.js]: Agree. You can keep the original.

**Pull request closed without merging**